Crypto/Cipher/ChaChaPoly1305.hs

```diff
@@ -81,6 +81,18 @@ instance ByteArrayAccess Nonce where
   withByteArray (Nonce8  n) = B.withByteArray n
   withByteArray (Nonce12 n) = B.withByteArray n
 
+instance Eq Nonce where
+  (Nonce8  a) == (Nonce8  b) = a == b
+  (Nonce12 a) == (Nonce12 b) = a == b
+  (Nonce8  a) == (Nonce12 b) = a == b
```

Comment:
instead of checking mismatching constructor, why not doing a `_ == _ = False`  directly ?